.github/workflows/pull_request.yml

```diff
@@ -93,18 +113,33 @@ jobs:
             sudo dpkg -i wkhtmltox_0.12.6-1.bionic_amd64.deb
       - name: Write app version
         run: printf "${{ github.event.pull_request.head.sha }}" > .application-version
-      - name: Install pipenv
-        run: pip install pipenv==2023.8.22
-      - name: Install virtual environment
-        run: pipenv install --dev
+      - name: Install poetry
+        uses: snok/install-poetry@v1
+        with:
+          version: 1.7.1
+          virtualenvs-create: true
+          virtualenvs-in-project: false
+          virtualenvs-path: ~/.virtualenvs
+      - name: Cache Poetry virtualenv
+        uses: actions/cache@v3
+        id: cache-virtualenv
+        with:
+          path: ~/.virtualenvs
+          key: ${{ runner.os }}-${{ env.PYTHON_VERSION }}-virtualenvs-${{ hashFiles('poetry.lock') }}
+      - name: Install Dependencies 
+        if: steps.cache-virtualenv.outputs.cache-hit != 'true'
+        run: poetry install --no-interaction
       - name: Load templates
         run: make load-design-system-templates
       - name: Compile translations
         run: make translate
       - name: Link env vars
-        run: ln -sf .development.env .env
+        run: ln -sf .development.env .env 
+      - name: Load .env File
+        uses: xom9ikk/dotenv@v2
```

Comment:
We should only use well maintained or recognised authors. This doesn't seem to be. It would be good if Actions doesn't differ from local. Can you not use the poetry dot env plugin instead?.

Reply:
Poetry dot env plugin has an issue here with CI/CD. I think i miss read your message last week with the plugin name. I think you meant poetry-plugin-dotenv . I will double check with that plugin instead

Reply:
Updated to use the Poetry plugin dotenv. Had to be separately installed since its a plugin and not part of the lock and toml file